services/core-api/app/api/document_generation/resources/explosives_permit_document.py

```diff
@@ -0,0 +1,108 @@
+import requests
+from flask import request, Response, stream_with_context
+from flask_restplus import Resource, marshal
+from werkzeug.exceptions import BadRequest, InternalServerError, BadGateway
+from app.extensions import api, cache
+
+from app.api.utils.resources_mixins import UserMixin
+from app.api.constants import EXPLOSIVES_PERMIT_DOCUMENT_DOWNLOAD_TOKEN
+from app.config import Config
+
+from app.api.mines.documents.models.mine_document import MineDocument
+from app.api.mines.explosives_permit.models.explosives_permit import ExplosivesPermit
+from app.api.mines.explosives_permit.models.explosives_permit_document_type import ExplosivesPermitDocumentType
+from app.api.mines.explosives_permit.models.explosives_permit_document_xref import ExplosivesPermitDocumentXref
+
+from app.api.services.document_generator_service import DocumentGeneratorService
+from app.api.services.document_manager_service import DocumentManagerService
+from app.api.mines.explosives_permit.response_models import EXPLOSIVES_PERMIT_DOCUMENT_MODEL
+
+
+class ExplosivesPermitDocumentResource(Resource, UserMixin):
```

Comment:
the filename should reflect the class name so explosives_permit_document_resource.py

Reply:
Agree, but I borrowed the same name format as the `now_document.py` file in the same folder. Will change.

Reply:
Ah, those should both be dealt with then